packages/c/sshnpd/src/handle_ssh_request.c

```diff
@@ -165,24 +168,25 @@ void handle_ssh_request(atclient *atclient, pthread_mutex_t *atclient_lock, sshn
   char *signing_algo_str = cJSON_GetStringValue(signing_algo);
 
   size_t valueolen = 0;
-  res = atchops_base64_decode((unsigned char *)signature_str, strlen(signature_str), (unsigned char *)value, strlen(value),
-                              &valueolen);
+  res = atchops_base64_decode((unsigned char *)signature_str, strlen(signature_str), (unsigned char *)buffer,
+                              strlen(buffer), &valueolen);
+
   if (res != 0) {
     atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atchops_base64_decode: %d\n", res);
     cJSON_Delete(envelope);
     cJSON_free(payloadstr);
-    free(value);
+    free(buffer);
     return;
   }
 
   res = verify_envelope_signature(requesting_atsign_publickey, (const unsigned char *)payloadstr,
-                                  (unsigned char *)value, hashing_algo_str, signing_algo_str);
+                                  (unsigned char *)buffer, hashing_algo_str, signing_algo_str);
+  free(buffer);
```

Comment:
free(buffer) as soon as we no longer need it, hence all the removals of free(value) (old variable name) below